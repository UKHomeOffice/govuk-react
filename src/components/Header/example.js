import React from "react";
import Component from ".";

export default () => <Component>Example</Component>;
