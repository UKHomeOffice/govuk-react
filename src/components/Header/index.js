// https://govuk-elements.herokuapp.com/typography/#typography-headings
// https://github.com/alphagov/govuk_frontend_toolkit/blob/master/stylesheets/_typography.scss
// https://github.com/alphagov/govuk-frontend/blob/master/src/globals/scss/core/_typography.scss
// https://github.com/alphagov/govuk_elements/blob/master/packages/govuk-elements-sass/public/sass/elements/_elements-typography.scss

import glamorous from "glamorous";
import React, { createElement } from "react";
import PropTypes from "prop-types";

import { MEDIA_QUERIES, NTA_LIGHT } from "../../constants/index";
import { LEVEL_SIZE, FONT_SIZES, LEVEL_TAG } from "../../constants/typography";

const Header = glamorous(({ level, children, ...props }) =>
  createElement(LEVEL_TAG[level], props, children)
)(
  {
    fontFamily: NTA_LIGHT,
    fontWeight: "bold",
    margin: 0
  },
  ({ level, size = LEVEL_SIZE[level] }) => ({
    fontSize: FONT_SIZES[size].mobile.fontSize,
    lineHeight: FONT_SIZES[size].mobile.lineHeight,
    marginBottom: FONT_SIZES[size].mobile.spacing,
    [MEDIA_QUERIES.LARGESCREEN]: {
      fontSize: FONT_SIZES[size].tablet.fontSize,
      lineHeight: FONT_SIZES[size].tablet.lineHeight,
      marginBottom: FONT_SIZES[size].tablet.spacing
    },
    [MEDIA_QUERIES.PRINT]: {
      fontSize: FONT_SIZES[size].print.fontSize
    }
  })
);

Header.defaultProps = {
  level: 1,
  size: undefined
};

Header.propTypes = {
  level: PropTypes.number,
  size: PropTypes.oneOf(Object.keys(FONT_SIZES))
};

export default Header;

export const H1 = props => <Header level={1} size={LEVEL_SIZE[1]} {...props} />;
export const H2 = props => <Header level={2} size={LEVEL_SIZE[2]} {...props} />;
export const H3 = props => <Header level={3} size={LEVEL_SIZE[3]} {...props} />;
export const H4 = props => <Header level={4} size={LEVEL_SIZE[4]} {...props} />;
export const H5 = props => <Header level={5} size={LEVEL_SIZE[5]} {...props} />;
export const H6 = props => <Header level={6} size={LEVEL_SIZE[6]} {...props} />;
