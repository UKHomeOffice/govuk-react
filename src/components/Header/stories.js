import React from "react";
import { storiesOf } from "@storybook/react";
import { action } from "@storybook/addon-actions";
import Header, { H1, H2, H3, H4, H5, H6 } from ".";

storiesOf("Header", module).add("default", () => <Header>Default</Header>);
storiesOf("Header", module).add("Levels 1-6", () => (
  <div>
    <Header level={1}>A 48px Bold heading</Header>
    <Header level={2}>A 36px Bold heading</Header>
    <Header level={3}>A 24px Bold heading</Header>
    <Header level={4}>A 19px Bold heading</Header>
    <Header level={5}>h5</Header>
    <Header level={6}>h6</Header>
  </div>
));
storiesOf("Header", module).add("Shortcuts 1-6", () => (
  <div>
    <H1>h1</H1>
    <H2>h2</H2>
    <H3>h3</H3>
    <H4>h4</H4>
    <H5>h5</H5>
    <H6>h6</H6>
  </div>
));
storiesOf("Header", module).add("Differing sizes", () => (
  <div>
    <Header level={6} size="XXLARGE">
      h6 with XXLARGE style
    </Header>
    <Header level={2} size="XSMALL">
      h2 with XSMALL style
    </Header>
    <H3 size="LARGE">h3 with size large</H3>
  </div>
));
storiesOf("Header", module).add("Props pass through", () => (
  <div>
    <Header onClick={action("header-click")}>Click me</Header>
  </div>
));
