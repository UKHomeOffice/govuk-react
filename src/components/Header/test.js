import React from "react";
import ReactDOM from "react-dom";
import { shallow, mount, render } from "enzyme";

import Header from "./";

describe("Header", () => {
  const example = "example";
  const wrapper = <Header>{example}</Header>;

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(wrapper, div);
  });

  it("matches snapshot", () => {
    expect(shallow(wrapper)).toMatchSnapshot(`enzyme.shallow`);
    expect(mount(wrapper)).toMatchSnapshot(`enzyme.mount`);
    expect(render(wrapper)).toMatchSnapshot(`enzyme.render`);
  });
});
