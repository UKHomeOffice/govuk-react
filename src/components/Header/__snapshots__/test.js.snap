// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot: enzyme.mount 1`] = `
<glamorous(unknown)
  level={1}
>
  <Component
    className="css-dcfn97"
    level={1}
  >
    <h1
      className="css-dcfn97"
    >
      example
    </h1>
  </Component>
</glamorous(unknown)>
`;

exports[`Header matches snapshot: enzyme.render 1`] = `
<h1
  class="css-dcfn97"
>
  example
</h1>
`;

exports[`Header matches snapshot: enzyme.shallow 1`] = `
<Component
  className="css-dcfn97"
  level={1}
>
  example
</Component>
`;
